keps/sig-network/2943-networkpolicy-status/README.md

```diff
@@ -0,0 +1,458 @@
+
+	// Specification of the desired behavior for this NetworkPolicy.
+	// +optional
+	Spec NetworkPolicySpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+
+  // Status is the current state of the NetworkPolicy.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	// +optional
+	Status NetworkPolicyStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+```
+
+* Add the `NetworkPolicyStatus` struct as following:
+```go
+// NetworkPolicyStatus describe the current state of the NetworkPolicy.
+type NetworkPolicyStatus struct {
+    // Conditions holds an array of metav1,Conditions that describe the state of the NetworkPolicy.
+  // When a CNI or other NetworkPolicy controller implementation parses this object,
+  // it must ensure that the `Accepted` Condition is updated with the parsing outcome.
+  // The Reason and Description must indicate the cause of the failure, if there was one.
+  
+  // (the Conditions field should also have a default that adds the `Accepted` Condition
+  // with a `status` of `Unknown`.)
+ Conditions []metav1.Conditon
+}
+```
+being metav1.Condition:
+
+```go
+type Condition struct {
+	Type string `json:"type" protobuf:"bytes,1,opt,name=type"`
+	// status of the condition, one of True, False, Unknown.
+	Status ConditionStatus `json:"status" protobuf:"bytes,2,opt,name=status"`
+	// observedGeneration represents the .metadata.generation that the condition was set based upon.
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	ObservedGeneration int64 `json:"observedGeneration,omitempty" protobuf:"varint,3,opt,name=observedGeneration"`
+	// lastTransitionTime is the last time the condition transitioned from one status to another.
+	// This should be when the underlying condition changed.  If that is not known, then using the time when the API field changed is acceptable.
+	// +required
+	LastTransitionTime Time `json:"lastTransitionTime" protobuf:"bytes,4,opt,name=lastTransitionTime"`
+	// reason contains a programmatic identifier indicating the reason for the condition's last transition.
+	// Producers of specific condition types may define expected values and meanings for this field,
+	// and whether the values are considered a guaranteed API.
+	// The value should be a CamelCase string.
+	// This field may not be empty.
+	Reason string `json:"reason" protobuf:"bytes,5,opt,name=reason"`
+	// message is a human readable message indicating details about the transition.
+	// This may be an empty string.
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=32768
+	Message string `json:"message" protobuf:"bytes,6,opt,name=message"`
+}
+```
+The above is an extraction from https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L1475-L1521
+
+As a starting we will have an additional default condition 
```

Comment:
see above comment.

Reply:
what about leaving this comment as "unresolved" and as you said before we decide during implementation?

I will change and add this, can revert otherwise.